README.md

```diff
@@ -1,138 +1,615 @@
+  @register((target, name) => {
+    Object.defineProperty(target, "name", {
+      get() { return this[`__internal_${name}`]; },
+      set() { this[`__internal_${name}`] = value; this.render(); },
+      configurable: true
+    });
+  })
+}
+```
+
+Note, further built-in decorators as in [NEXTBUILTINS.md](./NEXTBUILTINS.md) may provide a more direct and statically analyzable way to implement `@tracked` and avoid the use of `Object.defineProperty`. This version unfortunately relies on metaprogramming when the class is defined.
+
+### `@bound`
+
+The `@bound` decorator makes a method auto-bound: it will carry around the original `this` value when accessed as `this.method` and not immediately called. This behavior matches Python's semantics, and it's been found useful in the React ecosystem, which makes frequent use of passing functions around. Example usage:
+
+```mjs
+import { @bound } from "./bound.mjs";
+
+class Foo {
+  x = 1;
+
+  @bound method() { console.log(this.x); }
+
+  queueMethod() { setTimeout(this.method, 1000); }
+}
+
+new Foo().queueMethod();  // will log 1, rather than undefined
+```
+
+One possible implementation, based on `@register`:
+
+```mjs
+// bound.mjs
+export decorator @bound {
+  @register((target, name) => {
+    const method = target[name];
+    Object.defineProperty(target, name, {
+      get() {
+        const bound = method.bind(this);
+        Object.defineProperty(this, name, { value: bound, configurable: true });
+      }
+      configurable: true
+    });
+  })
+}
+```
+
+There are various appraoches to writing an auto-bound decorator, but ultimately, the most efficient way may be built into the JavaScript engine; see [NEXTBUILTINS.md](./NEXTBUILTINS.md) for discussion of a built-in `@bound` decorator, and the [bound-decorator](https://github.com/mbrowne/bound-decorator) repository for another approach.
+
+### `@callable`
+
+The `@callable` decorator makes it possible to invoke the class without `new`. When a class decorated with `@callable` is called, its static `call` method is invoked.
+
+```mjs
+import { @callable } from "./callable.mjs";
+
+@callable
+class MyDate {
+  static call(...args) { return Date(...args) }
+  constructor(...args) { return new Date(...args) }
+}
+```
+
+An implementation in terms of `@wrap`:
+
+```mjs
+// callable.mjs
+
+decorator @call(callback) {
+  @wrap(klass => {
+    function subclass(...args) {
+      if (new.target === undefined) {
+        return callback.call(klass, ...args);
+      } else {
+        return Reflect.construct(klass, args, new.target);
+      }
+    }
+    subclass.__proto__ = klass;
+    subclass.prototype.__proto__ = klass;
+    return subclass;
+  })
+}
+
+export decorator @callable {
+  @call(function(...args) { return this.call(...args); })
+}
+```
+
+Note that a decorator like `@call` could be considered for a future built-in decorator, in a way that avoids creating an additional subclass.
```

Comment:
does this implementation allow `instanceof` to work? (eg, `new MyDate() instanceof MyDate && MyDate() instanceof MyDate`)

Reply:
Yeah, it should work, depending on what you put in your static call method. In NEXTBUILTINS.md, I'll suggest `@call` as a built-in decorator which does things in an even nicer way.